frontend/dockerfile/parser/directives.go

```diff
@@ -110,6 +112,16 @@ func (d *DirectiveParser) ParseAll(data []byte) ([]*Directive, error) {
 // This allows for a flexible range of input formats, and appropriate syntax
 // selection.
 func DetectSyntax(dt []byte) (string, string, []Range, bool) {
+	return detectDirective(keySyntax, dt)
+}
+
+// DetectCheck returns the check options of provided Dockerfile.
+// Follows all the same rules as the syntax directive.
+func DetectCheck(dt []byte) (string, string, []Range, bool) {
```

Comment:
I understand this is modeled after `DetectSyntax` but `DetectCheck` is very confusing here. As a quick fix I think we can change the private `detectDirective` to public `ParseDirective` (called with "check") and figure out later how we clean up the `DetectSyntax` part. (To me, the solution that could make sense is `ParseDirectives() (map, err)` and removing `DetectSyntax` but that should be left for follow-up PR).